File: demo-test/src/test/java/io/fries/demo/test/cucumber/wiremock/DisabledMockConfiguration.java
package io.fries.demo.test.cucumber.wiremock;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

@Configuration
@Profile("!mock")
public class DisabledMockConfiguration {

    @Bean
    public MockServersFactory disabledMockServersFactory() {
        return world -> MockServers.none();
    }
}

File: demo-test/src/test/kotlin/io/fries/demo/test/cucumber/wiremock/DisabledMockConfiguration.kt
package io.fries.demo.test.cucumber.wiremock

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

@Configuration
@Profile("!mock")
class DisabledMockConfiguration {

    @Bean
    fun disabledMockServersFactory(): MockServersFactory = { MockServers() }
}